test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartMessageReportGenerator.cs

```diff
@@ -0,0 +1,319 @@
+            this.SenderSource = Preconditions.CheckNonWhiteSpace(senderSource, nameof(senderSource));
+            this.ReceiverSource = Preconditions.CheckNonWhiteSpace(receiverSource, nameof(receiverSource));
+            this.TestReportType = testReportType;
+            this.SenderTestResults = Preconditions.CheckNotNull(senderTestResults, nameof(senderTestResults));
+            this.ReceiverTestResults = Preconditions.CheckNotNull(receiverTestResults, nameof(receiverTestResults));
+            this.completedStatusHistogram = new Dictionary<HttpStatusCode, List<TimeSpan>>();
+        }
+
+        internal string TrackingId { get; }
+
+        internal string SenderSource { get; }
+
+        internal string ReceiverSource { get; }
+
+        internal TestReportType TestReportType { get; }
+
+        internal ITestResultCollection<TestOperationResult> SenderTestResults { get; }
+
+        internal ITestResultCollection<TestOperationResult> ReceiverTestResults { get; }
+
+        public async Task<ITestResultReport> CreateReportAsync()
+        {
+            Logger.LogInformation($"Generating report: {nameof(EdgeHubRestartMessageReport)} for [{this.SenderSource}] and [{this.ReceiverSource}]");
+
+            bool isIncrementalSeqeunce = true;
+            long previousSeqNum = 0;
+            ulong passedMessageCount = 0;
+            ulong senderMessageCount = 0;
+            ulong receiverMessageCount = 0;
+            bool hasSenderResult = true;
+            long senderSeqNum = 0;
+            bool hasReceiverResult = true;
+            long receiverSeqNum = 0;
+
+            (senderMessageCount, hasSenderResult, senderSeqNum) =
+                await this.MoveNextSenderResultAsync(senderMessageCount);
+
+            (receiverMessageCount, hasReceiverResult, receiverSeqNum) =
+                await this.MoveNextReceiverResultAsync(receiverMessageCount);
+
+            while (hasSenderResult && hasReceiverResult)
+            {
+                this.ValidateResult(
+                    this.SenderTestResults.Current,
+                    this.SenderSource,
+                    TestOperationResultType.EdgeHubRestartMessage.ToString());
+
+                this.ValidateResult(
+                    this.ReceiverTestResults.Current,
+                    this.ReceiverSource,
+                    TestOperationResultType.Messages.ToString());
+
+                if (receiverSeqNum > senderSeqNum)
+                {
+                    // Increment sender result to have the same seq as the receiver
+                    (senderMessageCount, hasSenderResult, senderSeqNum) = await this.IncrementSequenceNumberAsync(
+                        hasSenderResult,
+                        this.MoveNextSenderResultAsync,
+                        receiverSeqNum,
+                        senderMessageCount);
+                }
+
+                if (receiverSeqNum < senderSeqNum)
```

Comment:
This case should not be happened in message scenario.  The sender report seq number should be same as number of restart count.

Reply:
I agree but let's keep this here just in case.

Reply:
Without this logic, if this case really happens, I think the report will show all result not matched. right?  If that's the case, please remove this logic.